### PR TITLE
fix: frontend api errors

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
API Error Handling

### Description
This PR replaces the API Error handling in the frontend.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts [hunk 1]
```diff
@@ -25,8 +30,20 @@ export const expectFetch = (expectedUrl: string, requestStatusCode: number, expe
...
+const mapCodeToText = (code: number): string => {
+  switch (code) {
+    case 400:
+      return 'bad_request'
+    case 401:
+      return 'forbidden'
+    default:
+      return 'unknown code'
+  }
+}
```
Comment (ErikMichelson) [inline frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts:49]: Do we want to maintain this list here?

**Review by DerMolly — APPROVED** [review]

File: frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts (hunk 1 above)
Comment (mrdrogdrog) [reply frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts:49]: its only relevant for tests

Comment (ErikMichelson) [reply frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts:49]: Still, this looks a bit misplaced and could be extracted into an own file.

Comment (mrdrogdrog) [reply frontend/src/api/common/api-request-builder/test-utils/expect-fetch.ts:49]: why? it's only used in the fetch mock to simulate the "statusText" property.

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**